br/pkg/storage/gcs.go

```diff
@@ -269,12 +270,22 @@ func (s *GCSStorage) URI() string {
 }
 
 // Create implements ExternalStorage interface.
-func (s *GCSStorage) Create(ctx context.Context, name string, _ *WriterOption) (ExternalFileWriter, error) {
-	object := s.objectName(name)
-	wc := s.bucket.Object(object).NewWriter(ctx)
-	wc.StorageClass = s.gcs.StorageClass
-	wc.PredefinedACL = s.gcs.PredefinedAcl
-	return newFlushStorageWriter(wc, &emptyFlusher{}, wc), nil
+func (s *GCSStorage) Create(ctx context.Context, name string, wo *WriterOption) (ExternalFileWriter, error) {
+	if wo == nil || wo.Concurrency <= 1 {
+		object := s.objectName(name)
+		wc := s.bucket.Object(object).NewWriter(ctx)
+		wc.StorageClass = s.gcs.StorageClass
+		wc.PredefinedACL = s.gcs.PredefinedAcl
+		return newFlushStorageWriter(wc, &emptyFlusher{}, wc), nil
+	}
+	uri := s.objectName(name)
+	w, err := NewGCSWriter(ctx, s.cli, uri, int(wo.PartSize), wo.Concurrency, s.gcs.Bucket)
```

Comment:
PartSize may be 0

Reply:
Same as S3

Reply:
S3 SDK can change 0 PartSize to the default value, but here we directly use it.